core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/CatalogMetaMapper.java

```diff
@@ -147,4 +148,11 @@ Integer updateCatalogMeta(
           + " SET deleted_at = UNIX_TIMESTAMP(CURRENT_TIMESTAMP(3)) * 1000.0"
           + " WHERE metalake_id = #{metalakeId} AND deleted_at = 0")
   Integer softDeleteCatalogMetasByMetalakeId(@Param("metalakeId") Long metalakeId);
+
+  @Delete(
+      "DELETE FROM "
+          + TABLE_NAME
+          + " WHERE deleted_at != 0 AND deleted_at < #{legacyTimeLine} LIMIT #{limit}")
```

Comment:
It's better to use `BETTWEN` or `> AND <` to take advantage of range queries for B-trees if we need create a index on this column.

Reply:
Good catch！Thx.